src/axolotl/core/trainer_builder/sft.py

```
@@ -0,0 +1,511 @@
+        if self.cfg.kd_top_k_before_softmax is not None:
+            training_arguments_kwargs["kd_top_k_before_softmax"] = (
+                self.cfg.kd_top_k_before_softmax
+            )
+
+        trainer_kwargs = {}
+
+        # Pop optimizer_cls_and_kwargs to trainer_kwargs
+        if hasattr(training_arguments_kwargs, "optimizer_cls_and_kwargs"):
+            trainer_kwargs["optimizer_cls_and_kwargs"] = getattr(
+                training_arguments_kwargs, "optimizer_cls_and_kwargs"
+            )
```

Ruff (0.8.2)

Do not call `getattr` with a constant attribute value. It is not any safer than normal property access.

Replace `getattr` with attribute access

(B009)